src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/OptionsTests.cs

```diff
@@ -1413,12 +1413,19 @@ public static void TypeInfoResolverChain_UpdateTypeInfoResolver_ResetsResolverCh
         }
 
         [Fact]
-        public static void TypeInfoResolverChain_AppendTypeInfoResolverToChain_ThrowsInvalidOperationException()
+        public static void TypeInfoResolverChain_AddsModifier_WorksAsExpected()
         {
+            // Regression test for https://github.com/dotnet/runtime/issues/112735
+            bool isModifierInvoked = false;
             var defaultResolver = new DefaultJsonTypeInfoResolver();
-            var options = new JsonSerializerOptions { TypeInfoResolver = defaultResolver };
+            var options = new JsonSerializerOptions { TypeInfoResolver = JsonContext.Default };
+            options.TypeInfoResolverChain.Add(new DefaultJsonTypeInfoResolver());
+            options.TypeInfoResolver = options.TypeInfoResolver!.WithAddedModifier(_ => isModifierInvoked = true);
+            options.MakeReadOnly();
+
+            options.GetTypeInfo(typeof(List<List<List<string>>>)); // A type not in JsonContext.Default
 
-            Assert.Throws<InvalidOperationException>(() => options.TypeInfoResolverChain.Add(options.TypeInfoResolver));
```

Comment:
There's no longer a situation where an exception is thrown? We're not missing a test now?

Reply:
It wasn't my intention to edit a test, I must have accidentally deleted it.

Reply:
Reinstated.